app/scripts/metamask-controller.js

```diff
@@ -3777,6 +3806,10 @@ export default class MetamaskController extends EventEmitter {
       engine.emit('notification', message),
     );
 
+    // BEGIN:ONLY_INCLUDE_IN(build-blockaid);
+    engine.push(createPPOMMiddleware(this.ppomController));
+    // END:ONLY_INCLUDE_IN(build-blockaid);
+
```

Comment:
Is there a specific reason for this to be the first middleware that we are pushing? I am wondering if the ppom middleware should run after the `createDupeReqFilterMiddleware` (ensuring that in MV3 we are not processing repeated requests). Also not sure if the ones to append `origin`, `tabId` and the `createLoggerMiddleware` shouldn't be added before.
What do you think?

Reply:
There is not specific reason for this to be first. It can go after all these middleware's. Let me update the PR.

Reply:
Done